pkg/statistics/handle/bootstrap.go

```diff
@@ -431,27 +431,86 @@ func (h *Handle) initStatsHistogramsByPaging(is infoschema.InfoSchema, cache sta
 	return nil
 }
 
-func (h *Handle) initStatsHistogramsConcurrently(is infoschema.InfoSchema, cache statstypes.StatsCache, totalMemory uint64, concurrency int) error {
-	var maxTid = maxTidRecord.tid.Load()
+type loadStrategy interface {
+	calculateTotalTaskCnt() uint64
+	generateAndSendTasks(worker *initstats.RangeWorker)
+}
+
+type maxTidStrategy struct {
+	maxTid int64
+}
+
+func newMaxTidStrategy() maxTidStrategy {
+	return maxTidStrategy{
+		maxTid: maxTidRecord.tid.Load(),
+	}
+}
+
+func (m maxTidStrategy) calculateTotalTaskCnt() uint64 {
+	maxTid := m.maxTid
+	intest.Assert(maxTid >= 0, "maxTid should be non-negative")
+	intest.Assert(maxTid == maxTidRecord.tid.Load(), "maxTid should be equal to the recorded maxTid")
+	totalTaskCnt := int64(1)
+	if maxTid > initStatsStep*2 {
+		totalTaskCnt = maxTid / initStatsStep
+	}
+	return uint64(totalTaskCnt)
+}
+
+func (m maxTidStrategy) generateAndSendTasks(worker *initstats.RangeWorker) {
 	tid := int64(0)
+	for tid <= m.maxTid {
+		worker.SendTask(initstats.Task{
+			StartTid: tid,
+			EndTid:   tid + initStatsStep,
+		})
+		tid += initStatsStep
+	}
+}
+
+type tableListStrategy struct {
+	tableIDs []int64
+}
+
+func newTableListStrategy(tableIDs []int64) tableListStrategy {
+	return tableListStrategy{
+		tableIDs: tableIDs,
+	}
+}
+
+func (t tableListStrategy) calculateTotalTaskCnt() uint64 {
+	intest.Assert(len(t.tableIDs) > 0, "tableIDs should not be empty")
+	return uint64(len(t.tableIDs))
+}
+
+func (t tableListStrategy) generateAndSendTasks(worker *initstats.RangeWorker) {
+	for _, tableID := range t.tableIDs {
+		worker.SendTask(initstats.Task{
+			StartTid: tableID,
+			EndTid:   tableID + 1,
+		})
```

Comment:
We are loading the stats table by table, not batch by batch.
Will this hurt the performance?

Reply:
For BR, I think they will only use the command without table options. If it is a partial backup and restore, we will fall back to the JSON approach.